gendarme-ignore.txt

```diff
@@ -553,7 +555,9 @@ M: Java.Interop.JniPeerMembers Java.Interop.JniPeerMembers::CreatePeerMembers(Sy
 R: Gendarme.Rules.Performance.AvoidUnusedParametersRule
 # This method is virtual; *overrides* need the parameter, not the default implementation.
 M: System.Collections.Generic.IEnumerable`1<System.Type> Java.Interop.JniRuntime/JniTypeManager::CreateGetTypesForSimpleReferenceEnumerator(System.String)
-
+# This method is retrieved by reflection in JniTypeManager::TryRegisterNativeMembers and called with the parameters. I guess we don't want to handle multiple signatures, because of these 2 methods.
+M: System.Void Java.Interop.JavaProxyObject::__RegisterNativeMembers(Java.Interop.JniType,System.String)
```

Comment:
We shouldn't hide these; we should *remove* these, as discussed on the call.

Instead of hard-coding `__RegisterNativeMembers` we should follow the spirit of [`[OnDeserialized]`](https://msdn.microsoft.com/en-us/library/system.runtime.serialization.ondeserializedattribute(v=vs.110).aspx) and:

1. Introduce a new custom attribute type, and
2. Require use of this custom attribute type on a `static` method which takes the same signature as `__RegisterNativeMembers`.

E.g. have a new `RegisterNativeMembersAttribute`, and have:

```csharp
partial class JavaProxyObject {
    [RegisterNativeMembers]
    static void RegisterNativeMembers (JniType type, string members)
    {
        // ...
    }
}
```

Reply:
This could (should?) be done as part of a separate PR.

Reply:
OK, lets do it in separate PR.

Reply:
Part of "let's do it in a separate PR" involves *not* hiding these methods in `gendarme-ignore.txt`. We don't want ti ignore them; we want to *fix* them.

It's not like this PR gets us down to 0 errors anyway, so there's no need to make further work (removing this change in the future PR).